pyanaconda/modules/security/certificates/installation.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os
+
+from pyanaconda.anaconda_loggers import get_module_logger
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.errors.installation import SecurityInstallationError
+from pyanaconda.core.path import make_directories, join_paths
+from pyanaconda.core.constants import PAYLOAD_TYPE_DNF, INSTALLATION_PHASE_PREINSTALL
+
+log = get_module_logger(__name__)
+
+
+class ImportCertificatesTask(Task):
+    """Task for importing certificates into a system.
+
+    Currently it just dumps the file to the path.
+    """
+
+    def __init__(self, sysroot, certificates, payload_type=None, phase=None):
+        """Create a new certificates import task.
+
+        :param str sysroot: a path to the root of the target system
+        :param certificates: list of certificate data holders
+        :param payload_type: a type of the payload
+        :param phase: installation phase - INSTALLATION_PHASE_PREINSTALL or None for any other
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._certificates = certificates
+        self._payload_type = payload_type
+        self._phase = phase
+
+    @property
+    def name(self):
+        return "Import CA certificates"
+
+    def _dump_certificate(self, cert, root):
+        """Dump the certificate into specified file and directory."""
+
+        if not cert.dir:
+            raise SecurityInstallationError(
+                "Certificate destination is missing for {}".format(cert.filename)
+            )
```

Comment:
Shouldn't there be a default path for certificate given by the pykickstart? I guess that most of the certificates will go to one place so we don't have to force this on users?

Reply:
What would be the default path?
At this point where we are just dumping a file to the path it does not make sense to me to have any.
I think even the tooling for certificate import is moving from

1) put the certificate to *the* directory (` /etc/pki/ca-trust/source/anchors/`)
2) run a tool - `update-ca-trust extract`

to

1) run `trust anchor <filename>`

(https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/securing_networks/using-shared-system-certificates_securing-networks#managing-trusted-system-certificates_using-shared-system-certificates)
So I think we don't want to own any default path.
It might be different if/when offer support to define where we should import the cert (like `--category global` or `--category edns`) but still ideally we would just pass the category to the import tool (`trust`). Something I drafted here: https://github.com/rvykydal/anaconda/commit/51c67c5a5b900b6b371d65ac3e125091580a2628

Reply:
Sounds good, so we can follow their example. Do we want to support only this even now?

I mean do we want the path solution instead of using `category`?

Reply:
@jkonecny12 Hmm. It may be worth considering removing the `--dir` solution completely. Depends if we feel comfortable offering the API. I think the `--dir` `--name` API is not very big burden given it just dumps the file, no more commitments (any more complex actions/imports would be done via `--category`). And it can provide some flexibility for clients when the things (paths, solutions) are not settled completely. If we offer only `--category`, (implementing only `--category dns` with the path and name of the cert baked in anaconda) at this point, adaptations to any changes of paths etc would require rebuild of Anaconda. Also it wouldn't be probably possible to  configure the destination via anaconda configuration as it is applied already in initramfs.

Reply:
@abbra do you think it would be feasible to provide only `--category dns` API by anaconda kickstart or should we rather offer also simple `--dir` and `--filename`? (Please see above).